### PR TITLE
Feat/alana riwayat

### PR DESCRIPTION
**Pull request by ARRARIAKU2**
## Describe your changes
feat: add riwayat transaksi, add user dashboard layout, add private provider, fix sidebar logic

## JIRA / Trello ticket number and link
https://trello.com/c/3oL7Egq0/47-fsw-order-history

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added tests.

## Does your PR have breaking change?
no

## Test Video / Screenshots
Screenshot (109)

### CODE REVIEW

**Review by pockypoem — APPROVED** [review]
RTD

File: src/pages/Login/login.hooks.tsx [hunk 1]
```diff
@@ -45,14 +45,22 @@ function LoginHooks() {
       if (email === "" || password === "") {
```
Comment (angga-22) [inline src/pages/Login/login.hooks.tsx:45]: if !email || !password @ARRARIAKU2

File: src/pages/Login/login.hooks.tsx [hunk 2]
```diff
@@ -78,19 +86,31 @@ function LoginHooks() {
...
-      setLoginError({
-        message: "Login Berhasil!",
-        status: "success",
-      });
-
-      localStorage.setItem("token", response?.data?.access_token);
+      if (response.data.access_token) {
+        setLoginError({
+          message: "Login Berhasil!",
+          status: "success",
```
Comment (angga-22) [inline src/pages/Login/login.hooks.tsx:92]: status bisa pake yg di folder constant @ARRARIAKU2

File: src/routes/routes.tsx [hunk 3]
```diff
@@ -31,7 +34,12 @@ function Router() {
             <Route path="/artikel/:id" element={<Detailartikel />} />
             <Route path="/checkout" element={<Checkout />} />
             <Route path="/invoices" element={<Invoice />} />
-            <Route path="/profile" element={<Profile />} />
+            <Route element={<LayoutDashboard />}>
+              <Route path="/profile" element={<Profile />} />
+              <Route path="/riwayat-transaksi" element={<Riwayat />} />
```
Comment (angga-22) [inline src/routes/routes.tsx:39]: wrap private route / yg wajib login untuk akses route nya, misal: /payment, /riwayat-transaksi, dan seterusnya @ARRARIAKU2

File: src/pages/Login/login.hooks.tsx (hunk 1 above)
Comment (ARRARIAKU2) [reply src/pages/Login/login.hooks.tsx]: done

File: src/pages/Login/login.hooks.tsx (hunk 2 above)
Comment (ARRARIAKU2) [reply src/pages/Login/login.hooks.tsx]: done

**Review by angga-22 — APPROVED** [review]
LGTM

**Pull request merged**